### ISSUE TITLE
[gatsby-plugin-page-creator] (Suggestion) Use explicitly plugin page creator for new gatsby sites

### ISSUE THREAD

**Issue by matthieuauger** [gatsbyjs/gatsby#15873]
## Summary

Hello 🙋‍♂️I'm currently working on the migration of a gatsby site to a gatsby theme. One of the issues I faced is [explained here](https://github.com/gatsbyjs/gatsby/blob/69445cec9b93897a5819f344965da57bfddd5f22/docs/docs/themes/converting-a-starter.md#sourcing-pages
). By default a gatsby site will load pages in src/pages directory. A theme doesn't and need to have the gatsby-page-creator plugin installed and enabled.

```
{
    resolve: `gatsby-plugin-page-creator`,
    options: {
        path: `${__dirname}/src/pages`,
    },
},
```

In order to have consistency and ease the migration of a site to a theme, could we explicitly add the plugin page creator in new sites created in Gatsby as well ?

**Comment by matthieuauger** [comment]
@ChristopherBiscardi As this is related to themes, you may have an opinion on this, your feedback would be really helpful 💯

**Comment by matthieuauger** [comment]
Hello @gatsbyjs/themes @gatsbyjs/core ✋Does it make sense to you? I'm working to ease the migration of gatsby sites to gatsby themes and this change could save a bit of time to gatsby theme creators.

If I'm correct, Gatsby already uses gatsby-plugin-page-creator under the hood and [seems loaded here](https://github.com/gatsbyjs/gatsby/blob/34e4add901738e2b15fd1b5875ccb76871ce57a3/packages/gatsby/src/bootstrap/load-plugins/load.js#L196)

I think we may explicitly expose the configuration in gatsby-config.js to ease theme creation

**Comment by ChristopherBiscardi** [comment]
https://github.com/gatsbyjs/gatsby/issues/15873

**Comment by sidharthachatterjee** [comment]
Published in `gatsby@2.15.6`
